src/core/qgsmaplayer.h

```diff
@@ -2090,6 +2101,8 @@ class CORE_EXPORT QgsMapLayer : public QObject
     QString generalHtmlMetadata() const;
 #endif
 
+    std::unique_ptr<QgsDataProvider> mPreloadedProvider;
```

Comment:
It is worth adding a comment here - that it is optionally used when loading a project, and then it is released...